### PR TITLE
better differentiation between client and server requests

### PR DESCRIPTION
**Pull request by iosifch**
Hi! I suggest some small modifications to the description of spans for better readability. Therefore, I've added the type of the request, whether it's client or server, along with the URL, in the description. What do you think?

### CODE REVIEW

**Review by beniamin — APPROVED** [review]

**Comment by brettmc** [comment]
I'm not convinced about this change. There is a little bit of guidance in https://github.com/open-telemetry/opentelemetry-specification/blob/v1.24.0/specification/trace/api.md#span and also note that we set an attribute for SpanKind to represent client/server request.
Do any of the other language implementations do this?

In passing, I also see that symfony requests use the full request URI in span name, which is high cardinality and not good (per above link). It should try to use a route name, or else fall back to a very generic "http <method>"

**Comment by iosifch** [comment]
While there's an attribute that represent the request type, I find that is very hard to distinguish between them, but it could okay without it.

About server requests: using the route name instead of URI looks better as the route name is more generic. As stated in the specs, the span name should not be too general so adding the route name seems to complain with the specs. 

Regarding client requests, I now have doubts about whether the URI should be added to the span's name because the URI could be too long and verbose. Any other ideas here?

**Comment by beniamin** [comment]
Regarding client requests I think it would be a good ideea to add only the requested hostname to the span name. This way we can increase the readability of the spans without having to expand it and look into all their attributes.

File: src/SymfonyInstrumentation.php [hunk 1]
```diff
@@ -79,6 +79,15 @@ public static function register(): void
                 $span = Span::fromContext($scope->context());
 
                 $request = ($params[0] instanceof Request) ? $params[0] : null;
+
+                if ($params[1] === HttpKernelInterface::MAIN_REQUEST) {
```
Comment (iosifch) [inline src/SymfonyInstrumentation.php:83]: Because only the main request has info about the route.

**Comment by iosifch** [comment]
I made a few changes as follows:
- only the main request span's name contains the route (the sub-requests are dispatched by twig or when a exception must be rendered)
- client requests contain, as @beniamin recommended, only the hostname

**Comment by brettmc** [comment]
> client requests contain, as @beniamin recommended, only the hostname

That seems ok to me, it's low cardinality and the uri is available as an attribute if more detail is required.

I added this to the SIG agenda - it would be good to be consistent across auto-instrumentation modules. And, if possible, be consistent with what other languages do.

**Comment by brettmc** [comment]
@iosifch I just noticed you've submitted this PR against a read-only subtree split. The main repo is https://github.com/open-telemetry/[opentelemetry-php-contrib](https://github.com/open-telemetry/opentelemetry-php-contrib)

I wondered why I didn't see it in our slack notifications, and now I know why...

**Comment by iosifch** [comment]
Oh, my bad. I'll make the pull request again. Maybe this one should be prefixed with READONLY?

**Comment by iosifch** [comment]
Oh, my bad. I'll make the pull request again. Maybe this repository should be prefixed with READONLY?

**Comment by brettmc** [comment]
> Maybe this repository should be prefixed with READONLY?

It's just been improved somewhat for the main repositories, in https://github.com/open-telemetry/opentelemetry-php/pull/1108
A follow-up to do the same to the contrib repo would be welcome.
There actually is a `[READONLY]` on the repository description, but it's fairly unhelpful as you can't see it outside of the org's repo list.

**Pull request closed without merging**